python/modules/objects.py

```diff
@@ -0,0 +1,201 @@
+    def dimensions(self):
+        return self._dimensions
+
+    @property
+    def name(self):
+        return self._name
+
+
+class Mobile(Object3d):
+
+    def __init__(self, name, dimensions):
+        """
+        Mobile class constructor.
+        """
+        Objet3d.__init__(self, name, dimensions)
+
+    def move(self, destination):
+        """
+        Move a mobile.
+        """
+        # change position
+        self.position.move(destination)
+        # refresh the vertices
+        self.vertices = pos_to_vertices(self.position, self.dimensions)
+
+
+class Hangar(Object3d):
+
+    def __init__(self, name, dimensions):
+        """
+        Hangar class constructor.
+        """
+        Objet3d.__init__(self, name, dimensions)
+
+
+class Coord3d:
+
+    def __init__(self, coord=[0,0,0]):
+        """
+        Coord3d class constructor.
+        """
+        x, y, z = coord
+        self.x = x
+        self.y = y
+        self.z = z
+        self.spatial = [x, y, z]
+
+    def __str__(self):
+        """
+        Prettily display a Coord3d object.
+        """
+        return "({0:.3f}, {1:.3f}, {2:.3f})".format(self.x, self.y, self.z)
+
+    def move(self, destination):
+        """
+        Change the coordinates.
+        """
+        self.__init__(destination)
+
+
+class Coord6d:
+
+    def __init__(self, coord=[0,0,0,0,0,0]):
+        """
+        Coord6d class constructor.
+        """
+        x, y, z, alpha, beta, gamma = coord
+        self.x = x
+        self.y = y
+        self.z = z
+        self.spatial = [x, y, z]
+        self.alpha = alpha
+        self.beta = beta
+        self.gamma = gamma
+        self.angular = [alpha, beta, gamma]
+
+    def __str__(self):
+        """
+        Prettily display a Coord6d object.
+        """
+        return "({0:.3f}, {1:.3f}, {2:.3f}, {0:.3f}, {1:.3f}, {2:.3f})".format(
+                self.x, self.y, self.z, self.alpha, self.beta, self.gamma)
+
+    def move(self, destination):
+        """
+        Change the coordinates.
+        """
+        self.__init__(destination)
+
+    # def change_spatial(self, spatial):
+    #     """
+    #     Change the spatial coordinates.
+    #     """
+    #     self.spatial = spatial
+    #     self.x, self.y, self.z = spatial
+
+    # def change_angular(self, angular):
+    #     """
+    #     Change the angular coordinates.
+    #     """
+    #     self.angular = angular
+    #     self.alpha, self.beta, self.gamme = angular
+
+
+class Trajectory:
+
+    def __init__(self, name, array):
+        """
+        Trajectory class constructor.
+        """
+        self._name = name  # Reading access only
+        self.array = array
+
+    @property
+    def name(self):
+        """
+        Reading access to name.
+        """
+        return self._name
+
+    def __str__(self):
+        """
+        Prettily display a trajectory.
+        """
+        return "Trajectory :\n" + str(self.array)
+
+    def dx(self, nb_steps, j):
```

Comment:
What is this function supposed to do ?
I think it is supposed to return the actual step along the x direction that should be used to perform the movement, but that is just a guess.

Reply:
The function returns the infinitesimal distance along the x direction for the j-th interval of a trajectory.
I thought it would be useful to make the 'calculs.py' file more clear using these functions, but as you said in an object oriented style I think it will be better to leave the ugly formulas in the 'utils.py' file. If you don't see any problem with it, I will delete these methods.

Reply:
You can keep math related functions in your objects if you need to.

Just think about what methods you will need / want to use from the outside. It is fine to define too many methods at the beginning and remove them afterwards if you aren't too sure about it.

What I don't want you to do is define a lot of helper mathematical functions  that will work on the array contained in your objects because you did not define enough methods on your objects. That is to say, I don't want you to write a lot of methods which only use your object as a container for numpy arrays, if they need an object as argument then they should use it through its methods (be it decorated methods).

As a rule of thumb, things that take Objects as arguments (as in not a numpy array or other predefined Python types) should probably be a method. Your helper functions should almost only work on numpy arrays etc...

That is not written in stone though, and you might want to break the rule, depending on the context. I leave that to your appreciation :-)